### PR TITLE
feat(docker-compose): support entrypoint

### PR DESCRIPTION
**Pull request by Tucker-Eric**
Allows an entrypoint to be defined in a docker-compose service.

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/docker-compose/docker-compose-render.ts [hunk 1]
```diff
@@ -20,12 +20,13 @@ interface DockerComposeFileServiceSchema {
   readonly dependsOn?: string[];
   readonly build?: DockerComposeBuild;
   readonly image?: string;
-  readonly command?: string[];
+  readonly command?: string | string[];
```
Comment (mrgrain) [inline src/docker-compose/docker-compose-render.ts:23]:
```suggestion
  readonly command?: string[];
```

We can't have union types due, they don't work in other languages.

File: src/docker-compose/docker-compose-render.ts [hunk 2]
```diff
@@ -20,12 +20,13 @@ interface DockerComposeFileServiceSchema {
...
   readonly volumes?: DockerComposeVolumeMount[];
   readonly networks?: string[];
   readonly ports?: DockerComposeServicePort[];
   readonly environment?: Record<string, string>;
   readonly labels?: Record<string, string>;
+  readonly entrypoint?: string | string[];
```
Comment (mrgrain) [inline src/docker-compose/docker-compose-render.ts:29]:
```suggestion
  readonly entrypoint?: string[];
```
We can't have union types due, they don't work in other languages.

**Review by mrgrain — CHANGES_REQUESTED** [review]
Just the issue with the union types.

**Comment by mrgrain** [comment]
> and allows command and entrypoint to additionally be defined as support Docker's shell execution methods 

Not super sure what you mean by this 😅

File: src/docker-compose/docker-compose-render.ts (hunk 2 above)
Comment (Tucker-Eric) [reply src/docker-compose/docker-compose-render.ts:29]: Ahhh, this makes so much sense now!

**Comment by Tucker-Eric** [comment]
> > and allows command and entrypoint to additionally be defined as support Docker's shell execution methods
> 
> Not super sure what you mean by this 😅

Ahh, was supposed to support a string command/entrypoint for: 

image

but without union types it's not possible to support both.

**Review by mrgrain — APPROVED** [review]

**Comment by mrgrain** [comment]
@mergifyio update

**Comment by mrgrain** [comment]
@Tucker-Eric Our automation doesn't like your setup. 

> Mergify needs the author permission to update the base branch of the pull request.
https://github.com/gavel-io needs to [authorize modification on its head branch](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
err-code: F6886

If you can't would you mind re-raising the PR with your personal account?

Alternatively, if you branch is up to date with `main` I can merge manually, but depending on traffic you might have to do this multiple times.

**Pull request merged**